src/geo_index.c

```diff
@@ -5,28 +5,30 @@
 
 /* Add a docId to a geoindex key. Right now we just use redis' own GEOADD */
 int GeoIndex_AddStrings(GeoIndex *gi, t_docId docId, const char *slon, const char *slat) {
+  int rv = REDISMODULE_OK;
 
   RedisModuleString *ks = IndexSpec_GetFormattedKey(gi->ctx->spec, gi->sp, INDEXFLD_T_GEO);
   RedisModuleCtx *ctx = gi->ctx->redisCtx;
 
   /* GEOADD key longitude latitude member*/
   RedisModuleCallReply *rep = RedisModule_Call(ctx, "GEOADD", "sccl", ks, slon, slat, docId);
-  if (rep == NULL) {
-    return REDISMODULE_ERR;
-  }
+
   int repType = RedisModule_CallReplyType(rep);
-  RedisModule_FreeCallReply(rep);
-  if (repType == REDISMODULE_REPLY_ERROR) {
-    return REDISMODULE_ERR;
-  } else {
-    return REDISMODULE_OK;
+  if (rep == NULL || repType == REDISMODULE_REPLY_ERROR) {
+    rv = REDISMODULE_ERR;
   }
+
+  RedisModule_FreeCallReply(rep);
```

Comment:
If rep is NULL you will try to free NULL reply, not sure it will not cause errors

Reply:
return it above